src/segmented_cache.erl

```diff
@@ -275,15 +279,17 @@ handle_cast(_Msg, State) ->
 -spec handle_info(any(), #cache_state{}) -> {noreply, #cache_state{}}.
 handle_info(purge, #cache_state{ttl = TTL} = State) ->
     purge_last_segment_and_rotate(State),
-    TimerRef = erlang:send_after(TTL, self(), purge),
-    {noreply, State#cache_state{timer_ref = TimerRef}};
+    case TTL of
+        infinity -> {noreply, State};
+        _ -> {noreply, State#cache_state{timer_ref = erlang:send_after(TTL, self(), purge)}}
+    end;
 handle_info(_Msg, State) ->
     {noreply, State}.
 
 %% @private
 terminate(_Reason, #cache_state{scope = Scope, name = Name, timer_ref = TimerRef}) ->
     pg:leave(Scope, Name, self()),
-    erlang:cancel_timer(TimerRef),
+    undefined =/= TimerRef andalso erlang:cancel_timer(TimerRef),
```

Comment:
Minor, style: `andalso` is best used in cases when the result is not ignored - not just for side effects. Otherwise it can be confusing.

Reply:
Ah, bad habits 😅